src/Handler/CommandNameExtractor/NamedCommandExtractor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace League\Tactician\Handler\CommandNameExtractor;
+
+/**
+ * Extract the name from a NamedCommand
+ */
+class NamedCommandExtractor implements CommandNameExtractor
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getNameForCommand($command)
+    {
+        if ($command instanceof NamedCommand) {
+            return $command->getName();
+        }
+
+        // fallback to class name
+        return get_class($command);
```

Comment:
I'd rather it just throw an exception here, honestly. If you'd like to have a custom fallback scenario, I'd recommend doing that per app or creating a chained version.

We might want a standard `CouldNotDetermineCommandName` exception that all extractors can throw for situations like this.